### DISCUSSION TITLE
Threats manager Studio Feedback

### DISCUSSION THREAD

**Discussion by rsrinivasanhome** [simonec73/threatsmanager#54] [General]
I have been working with threats manager studio for almost a year and created a couple of threat models and found the following key benefits.  

- Takes less time to analyze threats as we have the option to select reusable mitigations instead of rewriting the mitigation in free text again and again
- Self check enabled by default which reduces threat modeling review time
- Reviews of threat model  are tracked
- Mitigation rules allows to generate a mitigation backlog example -  "Create a read-only key to access the database" etc
- Ability to document complex system(s).  You can link diagram(s) .  This is very helpful to represent complex systems .  I was surprised as I could not see this feature even in draw.io
- MTMT becomes slow when the template grows in threats manager studio we have not seen this behavior. 
- The same instance of the stencil can be used in multiple diagram and depict different functional flow(s)
- Threats manager studio user has a small learning curve but once you get used to it you find the GUI superior to MTMT . Ex -  You really know when a stencil is in a boundary.  Easier to create flows.

For me the following are area(s) which need to be fixed for more adoption

1)[ threat model gets cluttered on opening](https://github.com/simonec73/threatsmanager/issues/53) - 
2) undo
3) Once threat is generated and we try to change stencil property and regenerate old threats related to the changed property are not [deleted](https://github.com/simonec73/threatsmanager/issues/24)

For me 2 and 3 are OK as workarounds are simple .  Item 1 for me is the highest priority to fix .  The workaround takes more time and the issue occurs more often when creating a complex threat model .When is the next release planned? 
 Can the issue of threat model getting cluttered on opening be considered in it?

**Comment by simonec73** [comment]
Thank you very much for your feedback. I'm happy to confirm that points 1 and 2 have already been implemented and will be included in the upcoming release. I'm still testing the Undo/Redo capabilities to ensure they operate correctly. It's a major change, probably the most important one since I started working on TMS.
Point 3 is by design. If you change something, I'm not going to override it. You can already delete the existing Threat Events, which is usually too much. I am thinking to make it a little smarter to allow deleting just what hasn't been changed.
Consider that you can always remove the threats you want to be replaced and then trigger the automatic generation on the affected object.

I plan to release the new version in a few months. It will be before the end of the year, probably significantly earlier than that. From that point on, please expect more frequent releases.

**Comment by rsrinivasanhome** [comment]
@simonec73   I saw a new version of threats manager studio - 2.1.0

1)[ threat model gets cluttered on opening](https://github.com/simonec73/threatsmanager/issues/53) - is this closed in version 2.1.0 ?
2) undo  - I can  see  new undo button in version 2.1.0  
3) Once threat is generated and we try to change stencil property and regenerate old threats related to the changed property are not [deleted](https://github.com/simonec73/threatsmanager/issues/24)  - is this closed in version 2.1.0 ?

**Reply by simonec73** [reply]
Hi, I can confirm that both 1 and 2 are included. There are a few bugs I will fix with an upcoming release, which will happen very soon. I am not planning to address point 3, for now, as I specified previously.
